packages/typespec-ts/src/index.ts

```diff
@@ -94,6 +98,19 @@ export async function $onEmit(context: EmitContext) {
   const program: Program = context.program;
   const emitterOptions: EmitterOptions = context.options;
   const dpgContext = await createContextWithDefaultOptions(context);
+
+  if (
+    getFlavor(
+      emitterOptions,
+      getPackageDetails(dpgContext.program, emitterOptions)
+    ) !== "azure" &&
```

Comment:
Could you leverage isAzurePackage from rlc-common for this? thanks

Reply:
Also, could you move this logic to transformRLCOptions https://github.com/v-jiaodi/autorest.typescript/blob/193a08ba7036946362c03ca2146dab96da2bda18/packages/typespec-ts/src/transform/transfromRLCOptions.ts#L40 ?